### PR TITLE
MGMT-16612: notify when host updates status

### PR DESCRIPTION
**Pull request by rccrdpccl**
We were not notifying about status_info update in some cases, this PR addresses this issue.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/hostutil/update_host.go [hunk 1]
```diff
@@ -84,6 +85,22 @@ func UpdateHostStageTimeout(ctx context.Context, log logrus.FieldLogger, db *gor
 	return host, nil
 }
 
+func UpdateHostAndNotify(ctx context.Context, log logrus.FieldLogger, db *gorm.DB, stream stream.Notifier, infraEnvId strfmt.UUID,
+	hostId strfmt.UUID, srcStatus string, extra ...interface{}) (*common.Host, error) {
+	host, err := UpdateHost(log, db, infraEnvId, hostId, srcStatus, extra...)
+	if err != nil {
+		return nil, err
+	}
+	if host != nil {
```
Comment (adriengentil) [inline internal/host/hostutil/update_host.go:94]: is it possible to get `host = nil` if no error is returned from `UpdateHost`?

Comment (rccrdpccl) [reply internal/host/hostutil/update_host.go:94]: What I had in mind was an empty selection (where id = 'this_id_does_not_exist'). I am not too familiar with gorm but I'd say that shouldn't be an error?

Comment (adriengentil) [reply internal/host/hostutil/update_host.go:94]: look like there is a check for that https://github.com/rccrdpccl/assisted-service/blob/dce1b1958473c31b25aced54946a32ee2c2db16b/internal/host/hostutil/update_host.go#L90-L92 :thinking: 

Not a problem, I was wondering if `UpdateHost` guarantee if `host` is set when `err == nil`.

Comment (rccrdpccl) [reply internal/host/hostutil/update_host.go:94]: Nevermind, I've just looked it up and we use `Take`, which indeed returns error for empty result.

**Review by adriengentil — APPROVED** [review]

**Pull request merged**